### PR TITLE
[portsyncd] support read port configuration from configDB

### PR DESCRIPTION
**Pull request by ishidawataru**
**What I did**

read port configuration from configDB when `-p` is not specified.

**Why I did it**

configDB is the master data of port configuration.
We can simply use it instead of passing dedicated configuration file for `portsyncd`

**How I verified it**

Run `portsyncd` without `-p` option.

**Details if related**

### CODE REVIEW

File: portsyncd/portsyncd.cpp [hunk 1]
```diff
@@ -85,7 +86,11 @@ int main(int argc, char **argv)
         netlink.registerGroup(RTNLGRP_LINK);
         cout << "Listen to link messages..." << endl;
 
-        handlePortConfigFile(p, port_config_file);
+        if (port_config_file.empty()) {
+            handlePortConfigFile(p, port_config_file);
```
Comment (lguohan) [inline portsyncd/portsyncd.cpp]: use the port_config_file when it is empty?

File: portsyncd/portsyncd.cpp [hunk 2]
```diff
@@ -153,6 +158,35 @@ int main(int argc, char **argv)
...
+    cout << "Get port configuration from ConfigDB..." << endl;
+
+    Table table(&cfgDb, CFG_PORT_TABLE_NAME, CONFIGDB_TABLE_NAME_SEPARATOR);
+    std::vector<FieldValueTuple> ovalues;
+    std::vector<string> keys;
+    table.getKeys(keys);
+    for ( auto &k : keys ) {
+        table.get(k, ovalues);
+        vector<FieldValueTuple> attrs;
+        for ( auto &v : ovalues ) {
```
Comment (lguohan) [inline portsyncd/portsyncd.cpp]: please make all bracket into separate lines.

File: portsyncd/portsyncd.cpp (hunk 1 above)
Comment (ishidawataru) [reply portsyncd/portsyncd.cpp]: `string::empty()` returns 0 if the string is empty

http://www.cplusplus.com/reference/string/string/empty/

File: portsyncd/portsyncd.cpp (hunk 2 above)
Comment (ishidawataru) [reply portsyncd/portsyncd.cpp]: done

File: portsyncd/portsyncd.cpp [hunk 3]
```diff
@@ -85,7 +86,12 @@ int main(int argc, char **argv)
         netlink.registerGroup(RTNLGRP_LINK);
         cout << "Listen to link messages..." << endl;
 
-        handlePortConfigFile(p, port_config_file);
+        if (port_config_file.empty())
```
Comment (lguohan) [inline portsyncd/portsyncd.cpp]: true if the string length is 0, false otherwise.

Comment (ishidawataru) [reply portsyncd/portsyncd.cpp]: my bad. fixed

**Pull request merged**